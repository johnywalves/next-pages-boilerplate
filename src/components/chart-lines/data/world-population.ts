import { ChartLinesData } from '..'

const factor = Math.pow(10, 9)

const data: ChartLinesData = [
  {
    name: 'África',
    data: [
      { dimension: 2020, fact: 1360671810 / factor },
      { dimension: 2010, fact: 1055228072 / factor },
      { dimension: 2000, fact: 818946032 / factor },
      { dimension: 1990, fact: 638150629 / factor },
      { dimension: 1980, fact: 481536377 / factor },
      { dimension: 1970, fact: 365444348 / factor }
    ]
  },
  {
    name: 'América',
    data: [
      { dimension: 2020, fact: 1025766636 / factor },
      { dimension: 2010, fact: 935798901 / factor },
      { dimension: 2000, fact: 835703866 / factor },
      { dimension: 1990, fact: 718412840 / factor },
      { dimension: 1980, fact: 610082367 / factor },
      { dimension: 1970, fact: 508381762 / factor }
    ]
  },
  {
    name: 'Ásia',
    data: [
      { dimension: 2020, fact: 4663086535 / factor },
      { dimension: 2010, fact: 4220041327 / factor },
      { dimension: 2000, fact: 3735089604 / factor },
      { dimension: 1990, fact: 3210563577 / factor },
      { dimension: 1980, fact: 2635334228 / factor },
      { dimension: 1970, fact: 2144906290 / factor }
    ]
  },
  {
    name: 'Europa',
    data: [
      { dimension: 2020, fact: 745792196 / factor },
      { dimension: 2010, fact: 735613934 / factor },
      { dimension: 2000, fact: 726093423 / factor },
      { dimension: 1990, fact: 720320797 / factor },
      { dimension: 1980, fact: 692527159 / factor },
      { dimension: 1970, fact: 655923991 / factor }
    ]
  },
  {
    name: 'Oceânia',
    data: [
      { dimension: 2020, fact: 43933426 / factor },
      { dimension: 2010, fact: 37102764 / factor },
      { dimension: 2000, fact: 31222778 / factor },
      { dimension: 1990, fact: 26743822 / factor },
      { dimension: 1980, fact: 22920240 / factor },
      { dimension: 1970, fact: 19480270 / factor }
    ]
  }
]

export default data
