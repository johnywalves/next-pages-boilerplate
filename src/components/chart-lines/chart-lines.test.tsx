import { render } from '@testing-library/react'

import ChartLines from '.'
import data from './data/world-population'

describe('<ChartLines />', () => {
  it('should render chart of world population', () => {
    const { container } = render(<ChartLines chartData={data} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
