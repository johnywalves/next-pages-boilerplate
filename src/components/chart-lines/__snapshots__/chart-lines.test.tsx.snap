// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChartLines /> should render chart of world population 1`] = `
.c0 {
  display: block;
}

<div
  class="c0"
  id=":r0:"
>
  <svg
    height="0"
    width="0"
  >
    <g
      transform="translate(40,10)"
    >
      <g
        color="#fff"
        fill="none"
        font-family="sans-serif"
        font-size="0.75rem"
        font-weight="500"
        text-anchor="middle"
        transform="translate(0,-40)"
      >
        <path
          class="domain"
          d="M0.5,6V0.5H-59.5V6"
          stroke="currentColor"
        />
        <g
          class="tick"
          opacity="1"
          transform="translate(0.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            1970
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(-11.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            1980
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(-23.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            1990
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(-35.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            2000
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(-47.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            2010
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(-59.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            2020
          </text>
        </g>
      </g>
      <g
        color="#fff"
        fill="none"
        font-family="sans-serif"
        font-size="0.75rem"
        font-weight="500"
        text-anchor="end"
      >
        <path
          class="domain"
          d="M-6,-39.5H0.5V0.5H-6"
          stroke="currentColor"
        />
        <g
          class="tick"
          opacity="1"
          transform="translate(0,-39.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            0.0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,-35.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            0.5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,-31.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            1.0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,-27.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            1.5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,-23.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            2.0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,-19.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            2.5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,-15.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            3.0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,-11.500000000000002)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            3.5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,-7.499999999999998)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            4.0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,-3.499999999999999)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            4.5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,0.5)"
        >
          <line
            stroke="currentColor"
            x2="-6"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-9"
          >
            5.0
          </text>
        </g>
      </g>
      <path
        d="M-60,-29.115L-48,-31.558L-36,-33.448L-24,-34.895L-12,-36.148L0,-37.076"
        style="stroke: #e41a1c; stroke-width: 5; fill: none;"
      />
      <path
        d="M-60,-31.794L-48,-32.514L-36,-33.314L-24,-34.253L-12,-35.119L0,-35.933"
        style="stroke: #377eb8; stroke-width: 5; fill: none;"
      />
      <path
        d="M-60,-2.695L-48,-6.24L-36,-10.119L-24,-14.315L-12,-18.917L0,-22.841"
        style="stroke: #4daf4a; stroke-width: 5; fill: none;"
      />
      <path
        d="M-60,-34.034L-48,-34.115L-36,-34.191L-24,-34.237L-12,-34.46L0,-34.753"
        style="stroke: #984ea3; stroke-width: 5; fill: none;"
      />
      <path
        d="M-60,-39.649L-48,-39.703L-36,-39.75L-24,-39.786L-12,-39.817L0,-39.844"
        style="stroke: #ff7f00; stroke-width: 5; fill: none;"
      />
    </g>
  </svg>
</div>
`;
